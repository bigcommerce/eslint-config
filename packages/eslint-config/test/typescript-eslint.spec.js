import tseslint from '@typescript-eslint/eslint-plugin';
import tsParser from '@typescript-eslint/parser';
import { ESLint } from 'eslint';
import { join } from 'path';
import { fileURLToPath } from 'url';

import eslintSerializer from './eslint-serializer.js';

const __dirname = fileURLToPath(new URL('.', import.meta.url));

expect.addSnapshotSerializer(eslintSerializer);

async function getEslint() {
  return new ESLint({
    overrideConfigFile: true,
    baseConfig: [
      {
        files: ['**/*.ts'],
        languageOptions: {
          parser: tsParser,
          parserOptions: {
            project: join(process.cwd(), 'tsconfig.json'),
          },
        },
        plugins: {
          '@typescript-eslint': tseslint,
        },
        rules: {
          ...tseslint.configs.all.rules,
        },
      },
    ],
  });
}

describe('plugin:@typescript-eslint/all', () => {
  it('if this test fails, the plugin probably added new rules. Please check which rules were added/changed and add them to our config when appropriate', async () => {
    const eslint = await getEslint();
    const output = await eslint.calculateConfigForFile(join(__dirname, 'file.ts'));

    expect(output).toMatchSnapshot();
  });
});
