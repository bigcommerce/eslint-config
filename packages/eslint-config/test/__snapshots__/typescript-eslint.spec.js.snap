// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin:@typescript-eslint/all if this test fails, the plugin probably added new rules. Please check which rules were added/changed and add them to our config when appropriate 1`] = `
{
  "language": "@/js",
  "languageOptions": {
    "ecmaVersion": 2026,
    "parser": "typescript-eslint/parser@8.26.0",
    "parserOptions": {
      "project": "<rootDir>/tsconfig.json",
    },
    "sourceType": "module",
  },
  "linterOptions": {
    "reportUnusedDisableDirectives": 1,
  },
  "plugins": [
    "@",
    "@typescript-eslint:@typescript-eslint/eslint-plugin@8.26.0",
  ],
  "processor": undefined,
  "rules": {
    "@typescript-eslint/adjacent-overload-signatures": [
      2,
    ],
    "@typescript-eslint/array-type": [
      2,
    ],
    "@typescript-eslint/await-thenable": [
      2,
    ],
    "@typescript-eslint/ban-ts-comment": [
      2,
    ],
    "@typescript-eslint/ban-tslint-comment": [
      2,
    ],
    "@typescript-eslint/class-literal-property-style": [
      2,
    ],
    "@typescript-eslint/class-methods-use-this": [
      2,
    ],
    "@typescript-eslint/consistent-generic-constructors": [
      2,
    ],
    "@typescript-eslint/consistent-indexed-object-style": [
      2,
    ],
    "@typescript-eslint/consistent-return": [
      2,
      {
        "treatUndefinedAsUnspecified": false,
      },
    ],
    "@typescript-eslint/consistent-type-assertions": [
      2,
    ],
    "@typescript-eslint/consistent-type-definitions": [
      2,
    ],
    "@typescript-eslint/consistent-type-exports": [
      2,
    ],
    "@typescript-eslint/consistent-type-imports": [
      2,
    ],
    "@typescript-eslint/default-param-last": [
      2,
    ],
    "@typescript-eslint/dot-notation": [
      2,
      {
        "allowIndexSignaturePropertyAccess": false,
        "allowKeywords": true,
        "allowPattern": "",
        "allowPrivateClassPropertyAccess": false,
        "allowProtectedClassPropertyAccess": false,
      },
    ],
    "@typescript-eslint/explicit-function-return-type": [
      2,
    ],
    "@typescript-eslint/explicit-member-accessibility": [
      2,
    ],
    "@typescript-eslint/explicit-module-boundary-types": [
      2,
    ],
    "@typescript-eslint/init-declarations": [
      2,
    ],
    "@typescript-eslint/max-params": [
      2,
    ],
    "@typescript-eslint/member-ordering": [
      2,
    ],
    "@typescript-eslint/method-signature-style": [
      2,
    ],
    "@typescript-eslint/naming-convention": [
      2,
    ],
    "@typescript-eslint/no-array-constructor": [
      2,
    ],
    "@typescript-eslint/no-array-delete": [
      2,
    ],
    "@typescript-eslint/no-base-to-string": [
      2,
    ],
    "@typescript-eslint/no-confusing-non-null-assertion": [
      2,
    ],
    "@typescript-eslint/no-confusing-void-expression": [
      2,
    ],
    "@typescript-eslint/no-deprecated": [
      2,
    ],
    "@typescript-eslint/no-dupe-class-members": [
      2,
    ],
    "@typescript-eslint/no-duplicate-enum-values": [
      2,
    ],
    "@typescript-eslint/no-duplicate-type-constituents": [
      2,
    ],
    "@typescript-eslint/no-dynamic-delete": [
      2,
    ],
    "@typescript-eslint/no-empty-function": [
      2,
      {
        "allow": [],
      },
    ],
    "@typescript-eslint/no-empty-object-type": [
      2,
    ],
    "@typescript-eslint/no-explicit-any": [
      2,
    ],
    "@typescript-eslint/no-extra-non-null-assertion": [
      2,
    ],
    "@typescript-eslint/no-extraneous-class": [
      2,
    ],
    "@typescript-eslint/no-floating-promises": [
      2,
    ],
    "@typescript-eslint/no-for-in-array": [
      2,
    ],
    "@typescript-eslint/no-implied-eval": [
      2,
    ],
    "@typescript-eslint/no-import-type-side-effects": [
      2,
    ],
    "@typescript-eslint/no-inferrable-types": [
      2,
    ],
    "@typescript-eslint/no-invalid-this": [
      2,
      {
        "capIsConstructor": true,
      },
    ],
    "@typescript-eslint/no-invalid-void-type": [
      2,
    ],
    "@typescript-eslint/no-loop-func": [
      2,
    ],
    "@typescript-eslint/no-magic-numbers": [
      2,
    ],
    "@typescript-eslint/no-meaningless-void-operator": [
      2,
    ],
    "@typescript-eslint/no-misused-new": [
      2,
    ],
    "@typescript-eslint/no-misused-promises": [
      2,
    ],
    "@typescript-eslint/no-misused-spread": [
      2,
    ],
    "@typescript-eslint/no-mixed-enums": [
      2,
    ],
    "@typescript-eslint/no-namespace": [
      2,
    ],
    "@typescript-eslint/no-non-null-asserted-nullish-coalescing": [
      2,
    ],
    "@typescript-eslint/no-non-null-asserted-optional-chain": [
      2,
    ],
    "@typescript-eslint/no-non-null-assertion": [
      2,
    ],
    "@typescript-eslint/no-redeclare": [
      2,
    ],
    "@typescript-eslint/no-redundant-type-constituents": [
      2,
    ],
    "@typescript-eslint/no-require-imports": [
      2,
    ],
    "@typescript-eslint/no-restricted-imports": [
      2,
    ],
    "@typescript-eslint/no-restricted-types": [
      2,
    ],
    "@typescript-eslint/no-shadow": [
      2,
    ],
    "@typescript-eslint/no-this-alias": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-boolean-literal-compare": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-condition": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-parameter-property-assignment": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-qualifier": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-template-expression": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-type-arguments": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-type-assertion": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-type-constraint": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-type-parameters": [
      2,
    ],
    "@typescript-eslint/no-unsafe-argument": [
      2,
    ],
    "@typescript-eslint/no-unsafe-assignment": [
      2,
    ],
    "@typescript-eslint/no-unsafe-call": [
      2,
    ],
    "@typescript-eslint/no-unsafe-declaration-merging": [
      2,
    ],
    "@typescript-eslint/no-unsafe-enum-comparison": [
      2,
    ],
    "@typescript-eslint/no-unsafe-function-type": [
      2,
    ],
    "@typescript-eslint/no-unsafe-member-access": [
      2,
    ],
    "@typescript-eslint/no-unsafe-return": [
      2,
    ],
    "@typescript-eslint/no-unsafe-type-assertion": [
      2,
    ],
    "@typescript-eslint/no-unsafe-unary-minus": [
      2,
    ],
    "@typescript-eslint/no-unused-expressions": [
      2,
      {
        "allowShortCircuit": false,
        "allowTaggedTemplates": false,
        "allowTernary": false,
      },
    ],
    "@typescript-eslint/no-unused-vars": [
      2,
    ],
    "@typescript-eslint/no-use-before-define": [
      2,
    ],
    "@typescript-eslint/no-useless-constructor": [
      2,
    ],
    "@typescript-eslint/no-useless-empty-export": [
      2,
    ],
    "@typescript-eslint/no-wrapper-object-types": [
      2,
    ],
    "@typescript-eslint/non-nullable-type-assertion-style": [
      2,
    ],
    "@typescript-eslint/only-throw-error": [
      2,
    ],
    "@typescript-eslint/parameter-properties": [
      2,
    ],
    "@typescript-eslint/prefer-as-const": [
      2,
    ],
    "@typescript-eslint/prefer-destructuring": [
      2,
    ],
    "@typescript-eslint/prefer-enum-initializers": [
      2,
    ],
    "@typescript-eslint/prefer-find": [
      2,
    ],
    "@typescript-eslint/prefer-for-of": [
      2,
    ],
    "@typescript-eslint/prefer-function-type": [
      2,
    ],
    "@typescript-eslint/prefer-includes": [
      2,
    ],
    "@typescript-eslint/prefer-literal-enum-member": [
      2,
    ],
    "@typescript-eslint/prefer-namespace-keyword": [
      2,
    ],
    "@typescript-eslint/prefer-nullish-coalescing": [
      2,
    ],
    "@typescript-eslint/prefer-optional-chain": [
      2,
    ],
    "@typescript-eslint/prefer-promise-reject-errors": [
      2,
    ],
    "@typescript-eslint/prefer-readonly": [
      2,
    ],
    "@typescript-eslint/prefer-readonly-parameter-types": [
      2,
    ],
    "@typescript-eslint/prefer-reduce-type-parameter": [
      2,
    ],
    "@typescript-eslint/prefer-regexp-exec": [
      2,
    ],
    "@typescript-eslint/prefer-return-this-type": [
      2,
    ],
    "@typescript-eslint/prefer-string-starts-ends-with": [
      2,
    ],
    "@typescript-eslint/promise-function-async": [
      2,
    ],
    "@typescript-eslint/related-getter-setter-pairs": [
      2,
    ],
    "@typescript-eslint/require-array-sort-compare": [
      2,
    ],
    "@typescript-eslint/require-await": [
      2,
    ],
    "@typescript-eslint/restrict-plus-operands": [
      2,
    ],
    "@typescript-eslint/restrict-template-expressions": [
      2,
    ],
    "@typescript-eslint/return-await": [
      2,
    ],
    "@typescript-eslint/strict-boolean-expressions": [
      2,
    ],
    "@typescript-eslint/switch-exhaustiveness-check": [
      2,
    ],
    "@typescript-eslint/triple-slash-reference": [
      2,
    ],
    "@typescript-eslint/typedef": [
      2,
    ],
    "@typescript-eslint/unbound-method": [
      2,
    ],
    "@typescript-eslint/unified-signatures": [
      2,
    ],
    "@typescript-eslint/use-unknown-in-catch-callback-variable": [
      2,
    ],
    "class-methods-use-this": [
      0,
      {
        "enforceForClassFields": true,
        "exceptMethods": [],
        "ignoreOverrideMethods": false,
      },
    ],
    "consistent-return": [
      0,
      {
        "treatUndefinedAsUnspecified": false,
      },
    ],
    "default-param-last": [
      0,
    ],
    "dot-notation": [
      0,
      {
        "allowKeywords": true,
        "allowPattern": "",
      },
    ],
    "init-declarations": [
      0,
    ],
    "max-params": [
      0,
    ],
    "no-array-constructor": [
      0,
    ],
    "no-dupe-class-members": [
      0,
    ],
    "no-empty-function": [
      0,
      {
        "allow": [],
      },
    ],
    "no-implied-eval": [
      0,
    ],
    "no-invalid-this": [
      0,
      {
        "capIsConstructor": true,
      },
    ],
    "no-loop-func": [
      0,
    ],
    "no-magic-numbers": [
      0,
    ],
    "no-redeclare": [
      0,
      {
        "builtinGlobals": true,
      },
    ],
    "no-restricted-imports": [
      0,
    ],
    "no-return-await": [
      0,
    ],
    "no-shadow": [
      0,
      {
        "allow": [],
        "builtinGlobals": false,
        "hoist": "functions",
        "ignoreFunctionTypeParameterNameValueShadow": true,
        "ignoreOnInitialization": false,
        "ignoreTypeValueShadow": true,
      },
    ],
    "no-throw-literal": [
      0,
    ],
    "no-unused-expressions": [
      0,
      {
        "allowShortCircuit": false,
        "allowTaggedTemplates": false,
        "allowTernary": false,
        "enforceForJSX": false,
        "ignoreDirectives": false,
      },
    ],
    "no-unused-vars": [
      0,
    ],
    "no-use-before-define": [
      0,
      {
        "allowNamedExports": false,
        "classes": true,
        "enums": true,
        "functions": true,
        "ignoreTypeReferences": true,
        "typedefs": true,
        "variables": true,
      },
    ],
    "no-useless-constructor": [
      0,
    ],
    "prefer-destructuring": [
      0,
    ],
    "prefer-promise-reject-errors": [
      0,
      {
        "allowEmptyReject": false,
      },
    ],
    "require-await": [
      0,
    ],
  },
}
`;
